openmeter/billing/adapter.go

```diff
@@ -0,0 +1,162 @@
+package billing
+
+import (
+	"context"
+	"errors"
+	"fmt"
+)
+
+type TxAdapter interface {
+	ProfileAdapter
+
+	Commit() error
+	Rollback() error
+}
+
+type Adapter interface {
+	ProfileAdapter
+
+	WithTx(context.Context) (TxAdapter, error)
+}
+
+type ProfileAdapter interface {
+	CreateProfile(ctx context.Context, input CreateProfileInput) (*Profile, error)
+	GetProfileByKeyOrID(ctx context.Context, input GetProfileByKeyOrIDInput) (*Profile, error)
+	GetProfileByID(ctx context.Context, input GetProfileByIDAdapterInput) (*Profile, error)
+	GetProfileByKey(ctx context.Context, input GetProfileByKeyAdapterInput) (*Profile, error)
```

Comment:
do profiles need a key? Latest discussion was that key mostly makes sense for frontend facing resources.

Reply:
I am fine dropping the key, but my understanding was that that's an external ID from the foreign system. If we can drop the key that will help a lot with the logic. WDYT?